File: app/applet/replace_words.py
import os
import re

def replace_in_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Dosen -> Teacher
    content = re.sub(r'\bDosen\b', 'Teacher', content)
    content = re.sub(r'\bdosen\b', 'teacher', content)
    content = re.sub(r'\bDOSEN\b', 'TEACHER', content)
    
    # Mahasiswa -> Student
    content = re.sub(r'\bMahasiswa\b', 'Student', content)
    content = re.sub(r'\bmahasiswa\b', 'student', content)
    content = re.sub(r'\bMAHASISWA\b', 'STUDENT', content)
    
    # Siswa -> Student
    content = re.sub(r'\bSiswa\b', 'Student', content)
    content = re.sub(r'\bsiswa\b', 'student', content)
    content = re.sub(r'\bSISWA\b', 'STUDENT', content)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

for root, dirs, files in os.walk('src'):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            replace_in_file(os.path.join(root, file))

print("Replacement complete.")

File: app/applet/replace_words.ts
import { readFileSync, writeFileSync } from 'fs';
import { readdirSync, statSync } from 'fs';
import { join } from 'path';

function replaceInFile(filepath: string) {
    let content = readFileSync(filepath, 'utf-8');
    
    // Dosen -> Teacher
    content = content.replace(/\bDosen\b/g, 'Teacher');
    content = content.replace(/\bdosen\b/g, 'teacher');
    content = content.replace(/\bDOSEN\b/g, 'TEACHER');
    
    // Mahasiswa -> Student
    content = content.replace(/\bMahasiswa\b/g, 'Student');
    content = content.replace(/\bmahasiswa\b/g, 'student');
    content = content.replace(/\bMAHASISWA\b/g, 'STUDENT');
    
    // Siswa -> Student
    content = content.replace(/\bSiswa\b/g, 'Student');
    content = content.replace(/\bsiswa\b/g, 'student');
    content = content.replace(/\bSISWA\b/g, 'STUDENT');

    writeFileSync(filepath, content, 'utf-8');
}

function processDirectory(dir: string) {
    const files = readdirSync(dir);
    for (const file of files) {
        const filepath = join(dir, file);
        if (statSync(filepath).isDirectory()) {
            processDirectory(filepath);
        } else if (filepath.endsWith('.tsx') || filepath.endsWith('.ts')) {
            replaceInFile(filepath);
        }
    }
}

processDirectory('src');
console.log('Replacement complete.');
